### DISCUSSION TITLE
APNG status

### DISCUSSION THREAD

**Discussion by MaceWindu** [saucecontrol/PhotoSauce#156] [Q&A]
I have a quick question. Does this note about memory corruption is still actual?

https://github.com/saucecontrol/PhotoSauce/tree/master/src/NativeCodecs/Libpng#animated-png-support

Do you have any link to original issue?

**Comment by saucecontrol** [comment]
Yeah, the APNG support for `libpng` is provided only as a patch file, maintained at https://sourceforge.net/projects/libpng-apng/, and it has multiple bugs.  It's probable that the way I use the codec was not anticipated by the author of the patch.

I was able to fix the decode side relatively quickly for my own builds, but I haven't had time to diagnose the encode bug(s).  I'm not sure whether it's worth the effort, or if that effort is better spent on building a managed codec (or improving the one from ImageSharp).

If you manage to find the issue, I'd be happy to accept a PR adding the fix to the [libpng patches](https://github.com/saucecontrol/PhotoSauce/blob/master/build/vcpkg/ports/pspng/pspng-customize-code.patch) I use for my builds.
